ext/TensorAlgebraGradedUnitRangesExt/TensorAlgebraGradedUnitRangesExt.jl

```diff
@@ -1,8 +1,29 @@
 module TensorAlgebraGradedUnitRangesExt
-using GradedUnitRanges: AbstractGradedUnitRange, tensor_product
-using TensorAlgebra: TensorAlgebra
+
+using GradedUnitRanges: AbstractGradedUnitRange, dual, tensor_product
+using GradedUnitRanges.BlockArrays: Block, blocklengths, blocksize
+using Random: AbstractRNG
+using TensorAlgebra: TensorAlgebra, random_unitary
 
 function TensorAlgebra.:⊗(a1::AbstractGradedUnitRange, a2::AbstractGradedUnitRange)
   return tensor_product(a1, a2)
 end
+
+function TensorAlgebra.dual(a::AbstractGradedUnitRange)
+  return dual(a)
+end
+
+function TensorAlgebra.random_unitary(
+  rng::AbstractRNG,
+  elt::Type,
+  ax::Tuple{AbstractGradedUnitRange},
+)
+  a = zeros(elt, dual.(ax)..., ax...)
```

Comment:
Given that we have a specific implementation for the graded ranges here anyways, we could just drop the `dual` in the main package definition altogether, such that `TensorAlgebra` is decoupled from the symmetry.

Reply:
This version is actually only for matrices, so unless we change conventions, it is needed in both places for the current implementation.

Reply:
Ah I missed that, thanks!